tsdb/tsi1/tag_block.go

```diff
@@ -248,8 +248,8 @@ func (itr *tagBlockValueIterator) Next() TagValueElem {
 
 	// Unmarshal next element & move data forward.
 	itr.e.unmarshal(itr.data)
-	itr.data = itr.data[itr.e.size:]
 	_ = wait(itr.limiter, itr.data[:itr.e.size])
+	itr.data = itr.data[itr.e.size:]
```

Comment:
I found this while doing an audit of the other checks. It's performing the slice after moving the byte slice forward when it should happen before.

Reply:
I don't follow the change here: this reslice location shouldn't matter for `wait`. The reslice (`itr.data = itr.data[itr.e.size:]`) isn't performing an access to the slice's underlying array. `unmarshal` does access `itr.data`'s array. 

I'd put it `_ = wait(itr.limiter, itr.data[:itr.e.size])` before `unmarshal`. It'll make the array "hot" for the code using the Next() method.

Reply:
I believe there's still a bounds check on the slice even if it isn't accessing it. That's what happened with the other code that panicked.